src/api/task/api/handler_task_api.go

```diff
@@ -46,3 +47,31 @@ func (dh *TaskHandler) Post(g *gin.Context) {
 
 	g.JSON(http.StatusOK, dto.FromModel(dm))
 }
+
+func (dh *TaskHandler) Delete(g *gin.Context) {
+	dp := dto.Task{}
+
+	i, err := strconv.ParseInt(g.Param("id"), 10, 64)
+	if err != nil {
+		g.AbortWithStatusJSON(http.StatusUnprocessableEntity, errors.ErrResponse{
+			Err:     err,
+			Message: "Cannot parse ID",
+		})
+		return
+	}
+
+	dp.ID = i
+
+	dm, err := dh.Service.Delete(g, dp.ToModel())
+
+	if err != nil {
+		g.AbortWithStatusJSON(http.StatusUnprocessableEntity, errors.ErrResponse{
+			Err:     err,
+			Message: "Cannot delete task",
+		})
+		return
+	}
+
+	g.JSON(http.StatusOK, dto.FromModel(dm))
```

Comment:
No devolveria la entidad, da a confusion ya que estamos eliminando. Quizas devolveria un string con `Delete OK` o algo asi

Reply:
+1 seria demasiado complejo devolver algo tipo `Successfully deleted Project <id> `?

Reply:
+1 seria demasiado complejo devolver algo tipo `Successfully deleted Project <id>` ?

Reply:
makes sense, igual, al ser un soft delete, esto siempre funciona, aunque tratemos de eliminar 2 veces el mismo task, responder N veces `Successfully deleted task <id>` me hace ruido tambien

Reply:
bah, podria meter un hook `BeforeDelete` que chequee si el ID no se elimino antes, pero me parece overkill

Reply:
El soft delete deberia ser un delete en cuanto a la api se refiere, digamos, si eliminas un projecto y despues volves a intentar eliminarlo lo mas logico es que se actue como si no hubiera existido el projecto. Esto, como dice alex, se haria checkeando con algun select * from Projects where id=<id> and deletedDate is null. Sino tambien se puede sumar el is null al where del delete y responder en base a lo que responda la base de datos. Si vos en el where del delete tenes el deletedDate is null en teoria la db te deberia responder de alguna forma que te indique que no se afecto nada en la tabla

Reply:
yo no me meteria en eso la verdad, en todo caso no tendria que pasar, si el front actualiza la lista de tasks existentes no tendrian que tratar de eliminar 2 veces la misma